### PR TITLE
Fixing protobuf generation and InstrumentationScope

### PR DESCRIPTION
**Pull request by brettmc**
switching over to using opentelemetry-proto's own mechanism for generating protobuf interfaces, and
updating interfaces to latest version which works with our grpc exporter (v0.14.0). A separate issue + PR will
be required to work out why grpc exporting doesn't work with v0.15.0+

fixes: #660

### CODE REVIEW

File: script/proto_gen.sh [hunk 1]
```diff
@@ -1,31 +1,34 @@
...
+DESTINATION_DIR=$(realpath "$(dirname "${BASH_SOURCE[0]}")")/../proto/otel
+GPBMETA_DIR="GPBMetadata"
+OTEL_DIR="Opentelemetry"
 REPO_DIR=opentelemetry-proto
 
-mkdir -p $DESTINATION_DIR
-rm -R ./$REPO_DIR
-git clone https://github.com/open-telemetry/$REPO_DIR
+cd "${DESTINATION_DIR}" || exit
+rm -rf ./${REPO_DIR}
```
Comment (bobstrecansky) [inline script/proto_gen.sh:8]: probably want a check for something here.  If $`{REPO_DIR}` ends up being empty or being inadvertently set some other way......

File: script/proto_gen.sh [hunk 2]
```diff
@@ -1,31 +1,34 @@
...
+      git checkout "${TAG}"
 
-    git checkout "${TAG}"
+      echo "${TAG}"
+  )
 
-    echo "$TAG"
+  echo "Generating protobuf files for version ${TAG} ..."
+  make gen-php
+  rm -rf ${GPBMETA_DIR} ${OTEL_DIR}
```
Comment (bobstrecansky) [inline script/proto_gen.sh:24]: Same in re: `rm -rf`

File: script/proto_gen.sh [hunk 3]
```diff
@@ -1,31 +1,34 @@
...
-protoc --proto_path=$REPO_DIR/ --php_out=$DESTINATION_DIR --grpc_out=$DESTINATION_DIR \
- --plugin=protoc-gen-grpc=usr/local/bin/grpc_php_plugin $(find $REPO_DIR/opentelemetry/proto -iname "*.proto")
-
-echo "$TAG" > $DESTINATION_DIR/VERSION
+echo "Copying generated source..."
+cp -r ${REPO_DIR}/gen/php/${GPBMETA_DIR} .
+cp -r ${REPO_DIR}/gen/php/${OTEL_DIR} .
 
+echo "Cleaning up..."
+rm -rf ./${REPO_DIR}
```
Comment (bobstrecansky) [inline script/proto_gen.sh:33]: Ditto in re: `rm -rf`

File: script/proto_gen.sh (hunk 1 above)
Comment (brettmc) [reply script/proto_gen.sh:8]: It's set a couple of lines up, so I _think_ this is safe. I've confirmed that it's portable enough to be run from / or /scripts and still work...

**Comment by brettmc** [comment]
setting this to draft, since I now think I can get it working with 0.16 - the issue looks to be the renaming of InstrumentationLibrary to Scope

**Review by bobstrecansky — APPROVED** [review]

**Pull request merged**